qp_shotgun/kneaddata/kneaddata.py

```diff
@@ -0,0 +1,272 @@
+
+    Returns
+    -------
+    samples: list of tup
+        list of 3-tuples with run prefix, sample name, fwd read fp, rev read fp
+
+    Raises
+    ------
+    ValueError
+        If the rev is not an empty list and the same length than fwd seqs
+        The prefixes of the run_prefix don't match the file names
+
+    Notes
+    -----
+    """
+
+    # sort forward seqs
+    forward_seqs.sort()
+
+    # check that rev seqs are same len
+    if reverse_seqs:
+        if len(forward_seqs) != len(reverse_seqs):
+            raise ValueError('Your reverse and forward files are of different '
+                             'length. Forward: %s. Reverse: %s.' %
+                             (', '.join(forward_seqs),
+                              ', '.join(reverse_seqs)))
+        reverse_seqs.sort()
+
+    # get run prefixes
+    # These are prefixes that should match uniquely to forward reads
+    # sn_by_rp is dict of samples keyed by run prefixes
+    sn_by_rp = get_sample_names_by_run_prefix(map_file)
+
+    # make pairings
+    samples = []
+    used_prefixes = []
+    for i, f_fp in enumerate(forward_seqs):
+        # f_fp is the fwd read filepath
+        f_fn = basename(f_fp)
+
+        # iterate over run prefixes and make sure only one matches
+        run_prefix = None
+        for rp in sn_by_rp:
+            if f_fn.startswith(rp) and run_prefix is None:
+                run_prefix = rp
+            elif f_fn.startswith(rp) and run_prefix is not None:
+                raise ValueError('Multiple run prefixes match this fwd read: '
+                                 '%s' % f_fn)
+
+        # make sure that we got one matching run prefix:
+        if run_prefix is None:
+            raise ValueError('No run prefix matching this fwd read: %s'
+                             % f_fn)
+
+        if run_prefix in used_prefixes:
+            raise ValueError('This run prefix matches multiple fwd reads: '
+                             '%s' % run_prefix)
+
+        # if we have reverse reads, make sure the matching pair also
+        # matches the run prefix:
+        if (reverse_seqs and not
+                basename(reverse_seqs[i]).startswith(run_prefix)):
+            raise ValueError('Reverse read does not match this run prefix. '
+                             'Run prefix: %s\nForward read: %s\n'
+                             'Reverse read: %s\n' %
+                             (run_prefix, f_fn, basename(reverse_seqs[i])))
+
+        used_prefixes.append(run_prefix)
+        # create the tuple for this read set
+        if reverse_seqs:
+            samples.append((run_prefix, sn_by_rp[run_prefix], f_fp,
+                            reverse_seqs[i]))
+        else:
+            samples.append((run_prefix, sn_by_rp[run_prefix], f_fp, None))
+
+    return(samples)
+
+
+def format_kneaddata_params(parameters):
+
+    params = []
+
+    for param in sorted(parameters):
+        value = parameters[param]
+
+        if str(value) == 'True':
```

Comment:
The issue I was trying to get around with this was problems with integer params of 1 and 0 causing incorrect behavior, but I can see that this is preferable as long as we can depend on Qiita passing True and not 'True' in the parameter dict.